From 101e328981ddc7c7c6601f9cbb7eb9b2de38ef79 Mon Sep 17 00:00:00 2001
From: Michel Salim <michel@fb.com>
Date: Mon, 26 Oct 2020 17:25:45 -0700
Subject: [PATCH] Allow overriding version

Summary:
Upstream doesn't currently guarantee ABI stability, we want to let
downstream distributions override the version themselves when they
update Wangle.

Differential Revision: D24551628

fbshipit-source-id: 75ccf4aed41bfa097f6d7c15af9b4e893468e8ba
---
 wangle/CMakeLists.txt | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/wangle/CMakeLists.txt b/wangle/CMakeLists.txt
index 8df7c76f..4d3ee858 100644
--- a/wangle/CMakeLists.txt
+++ b/wangle/CMakeLists.txt
@@ -14,7 +14,15 @@
 
 cmake_minimum_required(VERSION 3.0.2)
 
-project("wangle" VERSION 1.0.0 LANGUAGES CXX C)
+if (NOT DEFINED PACKAGE_VERSION)
+  set(PACKAGE_VERSION "1.0.0")
+endif()
+
+if (NOT DEFINED SO_VERSION)
+  set(SO_VERSION 1)
+endif()
+
+project("wangle" VERSION ${PACKAGE_VERSION} LANGUAGES CXX C)
 
 if (NOT DEFINED CPACK_GENERATOR)
   set(CPACK_GENERATOR "RPM")
@@ -145,7 +153,7 @@ endif()
 
 if (BUILD_SHARED_LIBS)
   set_target_properties(wangle
-    PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION 1)
+    PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${SO_VERSION})
 endif()
 
 target_include_directories(
